File: python/load.py
import json
import requests
from urllib.parse import urlparse

def parse(uri):
    return 'http://' + urlparse(uri).netloc

# print(parse('derpy://127.0.0.1:8000'))

def post(url, data):
    r = requests.post(url, json=data)
    res = r.json()
    if not res['ok']:
        raise Exception(res['error'])
    return res['value']


class Module:
    def __init__(self, base):
        self.__dict__['base'] = base
    def __call__(self, *args):
        return post(self.base + '/call', args)
    def __setattr__(self, key, value):
        post(self.base + '/setProp', [ key, value ])
    def __getattr__(self, key):
        if key == 'base':
            return self.__dict__['base']
        return post(self.base + '/getProp', key)


x = Module(parse('derpy://127.0.0.1:8000'))

x.fooz = 'bar'

File: node/load.js
'use strict';

const { URL } = require('url');
const { spawnSync } = require('child_process');

// derpy://<ip / hostname>:<pid>/<uuid>

const parse = uri =>
	(({ host }) => `http://${host}`)
	(new URL(uri));

const post = (url, data) => {
	const res = JSON.parse(spawnSync('node', [ 'req', url ], {
		cwd: __dirname,
		input: JSON.stringify(data),
		encoding: 'utf8'
	}).stdout.trim());
	if (!res.ok) {
		throw new Error(res.error);
	}
	return res.value;
};

const wrap = base => new Proxy(
	(...args) => post(`${base}/call/`, args),
	{
		// getPrototypeOf: () => { throw new TypeError('Not implemented'); },
		// setPrototypeOf: () => { throw new TypeError('Not implemented'); },
		// isExtensible: () => { throw new TypeError('Not implemented'); },
		// preventExtensions: () => { throw new TypeError('Not implemented'); },
		// getOwnPropertyDescriptor: () => { throw new TypeError('Not implemented'); },
		// defineProperty: () => { throw new TypeError('Not implemented'); },
		has: (t, prop) => post(`${base}/hasProp`, prop),
		get: (t, prop) => post(`${base}/getProp`, prop),
		set: (t, k, v) => post(`${base}/setProp`, [ k, v ]),
		// deleteProperty: () => { throw new TypeError('Not implemented'); },
		// ownKeys: () => { throw new TypeError('Not implemented'); },
		apply: (t, self, args) => post(`${base}/call/`, args)
		// construct: () => { throw new TypeError('Not implemented'); }
	});

const main = uri => wrap(parse(uri));

module.exports = main;

if (require.main === module) {
	const m = main(process.argv[2]);
	global.m = m;
	require('repl').start({ useGlobal: true });
}
